hazelcast/src/main/java/com/hazelcast/query/Predicates.java

```diff
@@ -29,77 +29,342 @@
+ * case-sensitive enum member name.
+ * </ul>
+ * <li>{@link java.math.BigInteger BigInteger} type
+ * <ul>
+ * <li>Numeric values are converted to {@link java.math.BigInteger BigInteger} by applying widening or narrowing
+ * conversion as described in JLS 5.1.2 Widening Primitive Conversions and 5.1.3 Narrowing Primitive Conversions.
+ * <li>Boolean {@code true} and {@code false} are converted to {@link java.math.BigInteger#ONE BigInteger.ONE}
+ * and {@link java.math.BigInteger#ZERO BigInteger.ZERO} respectively.
+ * <li>A value of any other type is converted to string using {@link Object#toString()}, which is interpreted as
+ * a base 10 representation. If the representation is invalid, {@link NumberFormatException} will be thrown.
+ * </ul>
+ * <li>{@link java.math.BigDecimal BigDecimal} type
+ * <ul>
+ * <li>Numeric value are converted to {@link java.math.BigDecimal BigDecimal} by applying widening conversion as
+ * described in JLS 5.1.2 Widening Primitive Conversions.
+ * <li>Boolean {@code true} and {@code false} are converted to {@link java.math.BigDecimal#ONE BigDecimal.ONE}
+ * and {@link java.math.BigDecimal#ZERO BigDecimal.ZERO} respectively.
+ * <li>A value of any other type is converted to string using {@link Object#toString()}, which is interpreted as
+ * a base 10 representation. If the representation is invalid, {@link NumberFormatException} will be thrown.
+ * </ul>
+ * <li>{@link java.sql.Timestamp SQL Timestamp} type
+ * <ul>
+ * <li>Date values are converted to {@link java.sql.Timestamp SQL Timestamp} by applying {@link Date#getTime()} on them.
+ * <li>String values are interpreted in the same way as {@link java.sql.Timestamp#valueOf(String)} does.
+ * {@link RuntimeException} wrapping {@link java.text.ParseException} is thrown for invalid representations.
+ * <li>Any numeric value is interpreted as the number of milliseconds since <i>January 1, 1970, 00:00:00 GMT</i> by
+ * applying widening or narrowing conversion as described in JLS 5.1.2 Widening Primitive Conversions and 5.1.3
+ * Narrowing Primitive Conversions.
+ * </ul>
+ * <li>{@link java.sql.Date SQL Date} type
+ * <ul>
+ * <li>String values are interpreted in the same way as {@link java.sql.Date#valueOf(String)} does.
+ * {@link RuntimeException} wrapping {@link java.text.ParseException} is thrown for invalid representations.
+ * <li>Any numeric value is interpreted as the number of milliseconds since <i>January 1, 1970, 00:00:00 GMT</i>
+ * by applying widening or narrowing conversion as described in JLS 5.1.2 Widening Primitive Conversions and
+ * 5.1.3 Narrowing Primitive Conversions.
+ * </ul>
+ * <li>{@link Date} type
+ * <ul>
+ * <li>String values are interpreted as having <i>EEE MMM dd HH:mm:ss zzz yyyy</i> format. {@link RuntimeException}
+ * wrapping {@link java.text.ParseException} is thrown for invalid representations.
+ * <li>Any numeric value is interpreted as the number of milliseconds since <i>January 1, 1970, 00:00:00 GMT</i> by
```

Comment:
Again, isn't this using `java.util.Date.parse(String)`? Or is it using custom parsing?

Reply:
fixed